ui/analyse/src/study/studyChapters.ts

```diff
@@ -134,28 +166,24 @@ export function view(ctrl: StudyCtrl): VNode {
         },
       },
     },
-    ctrl.chapters
-      .list()
-      .map((chapter, i) => {
-        const editing = ctrl.chapters.editForm.isEditing(chapter.id),
-          loading = ctrl.vm.loading && chapter.id === ctrl.vm.nextChapterId,
-          active = !ctrl.vm.loading && current && !ctrl.relay?.tourShow.active && current.id === chapter.id;
-        return h(
-          'div',
-          {
-            key: chapter.id,
-            attrs: { 'data-id': chapter.id },
-            class: { active, editing, loading, draggable: canContribute },
-          },
-          [
-            h('span', loading ? h('span.ddloader') : ['' + (i + 1)]),
-            h('h3', chapter.name),
-            chapter.ongoing ? h('ongoing', { attrs: { ...dataIcon(''), title: 'Ongoing' } }) : null,
-            !chapter.ongoing && chapter.res ? h('res', chapter.res) : null,
-            canContribute ? h('i.act', { attrs: dataIcon('') }) : null,
-          ]
-        );
-      })
+    [
+      ctrl.chapters.list().length > 7
+        ? h('input.chapter-filter', {
+            on: { input: filterChapters },
+            attrs: { value: filterChInputValue, label: 'filter chapters', placeholder: 'filter chapters' },
+            hook: bind('input', filterChapters, ctrl.redraw),
+          })
+        : null,
+    ]
+      .concat(
+        ctrl.chapters.list().length > 7
+          ? filterChInputValue === ''
+            ? mapChapters(ctrl.chapters.list())
+            : filteredChapters.length >= 1
+            ? mapChapters(filteredChapters)
+            : h('h3', 'no results')
```

Comment:
`No results` should probably again be capitalized. Also should probably be made translatable but that can be left for later (if it's even merged in the first place; I guess I'm mainly also pointing out stuff for the future).